### ISSUE TITLE
CRL list monitoring/mining

### ISSUE THREAD

**Issue by plaintextcity** [koenrh/russian-trusted-root-ca#1]
Just a suggestion that it might be useful to mirror the CRL list as it is published from one of the certificate distribution points.  Also it would be interesting to monitor if they update all of the CDP's in a timely fashion.

                    URL=http://rostelecom.ru/cdp/rootca_ssl_rsa2022.crl
                    URL=http://company.rt.ru/cdp/rootca_ssl_rsa2022.crl
                    URL=http://reestr-pki.ru/cdp/rootca_ssl_rsa2022.crl
                    
The CA does not appear to have OCSP infrastructure, and the certificates do not contain OCSP responder paths. A mature CA would have this, before even CT.  The CRL is published with a one month validity period (technically one month+1 second, because it's inclusive).  This is a violation of the CA/Browser Baseline Requirements  `4.9.7 CRL issuance frequency` "If the CA publishes a CRL, then the CA SHALL update and reissue CRLs at least once every seven days, and the value of the nextUpdate field MUST NOT be more than ten days beyond the value of the thisUpdate field." 

Also from the list of Revoked Certificates, none are in the collected leaf certificates. They have revoked 10 certificates and published 8 certificate revocation lists up to April 7.

```
openssl crl -inform der -in ./subca_ssl_rsa2022.crl -noout -text
...
Certificate Revocation List (CRL):
        Version 2 (0x1)
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = RU, O = The Ministry of Digital Development and Communications, CN = Russian Trusted Sub CA
        Last Update: Apr  7 14:30:38 2022 GMT
        Next Update: May  7 14:30:38 2022 GMT
        CRL extensions:
            X509v3 Authority Key Identifier:
                keyid:D1:E1:71:0D:0B:2D:81:4E:6E:8A:4A:8F:4C:23:B3:4C:5E:AB:69:0B

            X509v3 CRL Number:
                8
Revoked Certificates:
    Serial Number: 111056
        Revocation Date: Mar 11 09:43:54 2022 GMT
    Serial Number: 1110F2
        Revocation Date: Mar 22 11:42:24 2022 GMT
    Serial Number: 111104
        Revocation Date: Mar 22 11:42:40 2022 GMT
    Serial Number: 111181
        Revocation Date: Mar 23 13:20:03 2022 GMT
    Serial Number: 1111E1
        Revocation Date: Mar 23 13:20:24 2022 GMT
    Serial Number: 111201
        Revocation Date: Mar 23 13:28:10 2022 GMT
    Serial Number: 111202
        Revocation Date: Mar 23 13:27:53 2022 GMT
    Serial Number: 11128F
        Revocation Date: Mar 25 08:45:04 2022 GMT
    Serial Number: 111340
        Revocation Date: Mar 29 10:15:08 2022 GMT
    Serial Number: 1113F5
        Revocation Date: Mar 31 10:20:41 2022 GMT
```

**Comment by koenrh** [comment]
> Just a suggestion that it might be useful to mirror the CRL list as it is published from one of the certificate distribution points. Also it would be interesting to monitor if they update all of the CDP's in a timely fashion.

Good idea! I have added the CRLs for the root CA (3 and 7) and sub CA (8) to the [crls folder](https://github.com/koenrh/russian-trusted-root-ca). I will add the scripts later.

I wasn't able to access `http://rostelecom.ru/cdp/rootca_ssl_rsa2022.crl` even when using a VPN with an exit node in Russia.

> The CA does not appear to have OCSP infrastructure, and the certificates do not contain OCSP responder paths. A mature CA would have this, before even CT.

Haven't seen any signs that they are building that infrastructure. I did come across what appears to be a test CT log instance.
